### PR TITLE
chore(iast): ast deny list

### PR DESCRIPTION
**Pull request by avara1986**
Update the deny list of IAST AST patching. The objective of this PR is to improve the django startup applications

## Checklist
- [x] PR author has checked that all the criteria below are met
- The PR description includes an overview of the change
- The PR description articulates the motivation for the change
- The change includes tests OR the PR description describes a testing strategy
- The PR description notes risks associated with the change, if any
- Newly-added code is easy to change
- The change follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html)
- The change includes or references documentation updates if necessary
- Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist
- [x] Reviewer has checked that all the criteria below are met 
- Title is accurate
- All changes are related to the pull request's stated goal
- Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- Testing strategy adequately addresses listed risks
- Newly-added code is easy to change
- Release note makes sense to a user of the library
- If necessary, author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by juanjux — CHANGES_REQUESTED** [review]
Blocking for now. I don't see the logic of banning all these modules, some of them look like could have propagation if properly patched, so I want to discuss that first.

File: ddtrace/appsec/_iast/_ast/ast_patching.py [hunk 1]
```diff
@@ -26,23 +26,550 @@
...
     "flask",
     "werkzeug",
+    "aiohttp._helpers",
+    "aiohttp._http_parser",
+    "aiohttp._http_writer",
+    "aiohttp._websocket",
+    "aiohttp.log",
+    "aiohttp.tcp_helpers",
+    "asgiref",
+    "asgiref.current_thread_executor",
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_ast/ast_patching.py:36]: If we add `"asgiref"` to the denylist, we don't need to add the submodules. This applies to `asyncio`, `backports`, `boto3`, `charset_normalizer`, `concurrent` and `django` as well

File: ddtrace/appsec/_iast/_ast/ast_patching.py [hunk 2]
```diff
@@ -26,23 +26,550 @@
...
+    "chardet.sbcsgroupprober",
+    "charset_normalizer",
+    "charset_normalizer.assets",
+    "charset_normalizer.constant",
+    "charset_normalizer.version",
+    "click",
+    "cmath",
+    "colorama",
+    "concurrent",
+    "concurrent",
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_ast/ast_patching.py:125]:
```suggestion
    "concurrent",
```

**Review by juanjux — APPROVED** [review]

File: ddtrace/appsec/_iast/_ast/ast_patching.py [hunk 3]
```diff
@@ -26,30 +26,285 @@
...
+    "django.templatetags.static",
+    "django.templatetags.tz",
+    "django.test",
+    "django.urls",
+    "django.urls.base",
+    "django.urls.conf",
+    "django.urls.converters",
+    "django.urls.exceptions",
+    "django.urls.resolvers",
+    "django.urls.utils",
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_ast/ast_patching.py:271]: As mentioned before, we don't need to include submodules if the parent is included

```suggestion
    "django.urls",
```

File: ddtrace/appsec/_iast/_ast/ast_patching.py [hunk 4]
```diff
@@ -26,30 +26,285 @@
...
+    "django_filters.constants",
+    "django_filters.exceptions",
+    "django_filters.fields",
+    "django_filters.filters",
+    "django_filters.filterset",
+    "django_filters.rest_framework",
+    "django_filters.rest_framework.backends",
+    "django_filters.rest_framework.filters",
+    "django_filters.rest_framework.filterset",
+    "django_filters.utils",
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_ast/ast_patching.py:286]: Ditto
```suggestion
    "django_filters",
```

File: ddtrace/appsec/_iast/_ast/ast_patching.py [hunk 5]
```diff
@@ -26,30 +26,285 @@
...
+    "django_filters.widgets",
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_ast/ast_patching.py:287]: Ditto
```suggestion
    "django_filters",
```

File: ddtrace/appsec/_iast/_ast/ast_patching.py [hunk 6]
```diff
@@ -262,16 +262,13 @@
...
     "django.test",
-    "django.urls",
     "django.urls.base",
     "django.urls.conf",
     "django.urls.converters",
     "django.urls.exceptions",
     "django.urls.resolvers",
     "django.urls.utils",
     "django.utils",
-    "django.views",
```
Comment (gnufede) [inline ddtrace/appsec/_iast/_ast/ast_patching.py]: Is this change on purpose? It's not a change from the peer review

**Review by gnufede — APPROVED** [review]

**Pull request merged**